Fix incorrectly showing checksums on non-ETH blockchains

jonathansmirnoff opened the pull request:
This PR fixes the issue #5838. I did another PR (https://github.com/MetaMask/metamask-extension/pull/5973) but I have closed it because of the git's history didn't was clear.

Please @whymarrh, @danjm and @alextsg take a look of it.

Thanks!

jonathansmirnoff committed:
Fixed issue #5838

Update changelog file

alepc253 commented:
Really useful for RSK community!

frankiebee commented:
@jonathansmirnoff will merge once tests are fixed

frankiebee commented:
at the very least unit tests need to be fixed. If e2e test are still failing once unit tests are fixed i may take a look and see why

jonathansmirnoff commented:
@frankiebee thanks a lot for the review. The branch develop has problems with unit test, at least the version of the few days ago.
Do you know if someone fix it?

jonathansmirnoff commented:
@frankiebee thanks for your review. Also as @alepc253 said this PR is a big steps for RSK community.
I added an improvement in the destination address validation for non-eth network. I think now is better. Also I fixed the problem with the unit tests. 
The only issue that I couldn't resolve is the e2e tests, only one test is not passing from 20.

frankiebee reviewed (dismissed):
thanks for this

frankiebee commented on ui/app/util.js:
```diff
@@ -60,6 +60,15 @@ module.exports = {
+}
+
+function isEthNetwork (netId) {
+  if (!netId || netId === '1' || netId === '3' || netId === '4' || netId === '42') {
```
this will fix e2e tests

jonathansmirnoff commented:
@frankiebee I included the ganache net id. Also I did a rebase from develop. But the e2e tests still failing. I think this is not related to me changes. Maybe the issue is in develop.
The error that I have is:
```
    Send ETH from inside MetaMask
      1) starts a send transaction


  20 passing (43s)
  1 failing

  1) MetaMask
       Send ETH from inside MetaMask
         starts a send transaction:
     WebDriverError: Element <button class="button btn-primary btn--large page-container__footer-button"> is not clickable at point (674,601.5) because another element <header> obscures it
      at Object.throwDecodedError (node_modules/selenium-webdriver/lib/error.js:514:15)
      at parseHttpResponse (node_modules/selenium-webdriver/lib/http.js:519:13)
      at doSend.then.response (node_modules/selenium-webdriver/lib/http.js:441:30)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)
  From: Task: WebElement.click()
      at thenableWebDriverProxy.schedule (node_modules/selenium-webdriver/lib/webdriver.js:807:17)
      at WebElement.schedule_ (node_modules/selenium-webdriver/lib/webdriver.js:2010:25)
      at WebElement.click (node_modules/selenium-webdriver/lib/webdriver.js:2092:17)
      at Context.<anonymous> (test/e2e/beta/metamask-beta-ui.spec.js:381:24)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)
```

tmashuang commented:
@jonathansmirnoff Seems I was mistaken on the default configuration between `ganache` and `ganache-cli`. We need to configure `ganache-cli network_id` in `package.json` to 5777. Please change 
https://github.com/MetaMask/metamask-extension/blob/develop/package.json#L41
 to
`"ganache:start": "ganache-cli --noVMErrorsOnRPCResponse -i 5777 -m 'phrase upgrade clock rough situate wedding elder clever doctor stamp excess tent'",
`

jonathansmirnoff commented:
Hi @frankiebee I did the changes. Thanks you again.

tmashuang commented:
@jonathansmirnoff Alright, this issue has been resolved in `develop` branch. Hopefully, one last update should get the CI to pass.

jonathansmirnoff commented:
@tmashuang you were right! All the tests passed!
@frankiebee I think we are good to go! 👍

jonathansmirnoff commented:
Hi @frankiebee, Do you know when you plan to merge this PR?

jonathansmirnoff commented:
@frankiebee I updated to the lasted version of develop. Also I improved the github history

frankiebee committed:
Merge branch 'develop' into develop
